addons/mail/static/src/models/thread_view_topbar/thread_view_topbar.js

```diff
@@ -147,10 +153,61 @@ function factory(dependencies) {
             this.messaging.models['mail.message'].unstarAll();
         }
 
+        /**
+         * Handles click on the "current user info".
+         *
+         * @param {MouseEvent} ev
+         */
+        onClickUserInfo(ev) {
+            if (!this.messaging.isCurrentUserGuest) {
+                return;
+            }
+            this.update({
+                doFocusOnGuestNameInput: true,
+                isEditingGuestName: true,
+                isMouseOverUserInfo: false,
+                pendingGuestName: this.messaging.currentGuest.name,
+            });
+        }
+
+        /**
+         * Handles click on the guest name.
+         *
+         * @param {MouseEvent} ev
+         */
+        onClickUserName(ev) {
+            if (!this.messaging.isCurrentUserGuest) {
+                return;
+            }
+            const selection = window.getSelection();
+            this.update({
+                doFocusOnGuestNameInput: true,
+                doSetSelectionDirectionOnGuestNameInput: selection.anchorOffset < selection.focusOffset ? 'forward' : 'backward',
+                doSetSelectionEndOnGuestNameInput: Math.max(selection.focusOffset, selection.anchorOffset),
+                doSetSelectionStartOnGuestNameInput: Math.min(selection.focusOffset, selection.anchorOffset),
+                isEditingGuestName: true,
+                isMouseOverUserInfo: false,
+                pendingGuestName: this.messaging.currentGuest.name,
+            });
+        }
+
         /**
          * Handles OWL update on this top bar component.
          */
         onComponentUpdate() {
+            if (this.doFocusOnGuestNameInput) {
+                this.guestNameInputRef.el.focus();
+                this.update({ doFocusOnGuestNameInput: clear() });
+                if (
+                    !this.doSetSelectionStartOnGuestNameInput &&
+                    !this.doSetSelectionEndOnGuestNameInput &&
+                    !this.doSetSelectionDirectionOnGuestNameInput
+                ) {
+                    // place cursor at end of text
+                    const { length } = (this.pendingGuestName || '');
+                    this.guestNameInputRef.el.setSelectionRange(length, length);
+                }
```

Comment:
code not needed, placing at the end should be done with the "doSetSelection..." fields when necessary (and in practice it's impossible to enable it without actually clicking on it and placing the cursor somewhere)